syntax = "proto3";

package embedding;

service Embedder {
  rpc EmbedBatch (EmbedRequest) returns (EmbedResponse);
}

message EmbedRequest {
  repeated string user_ids = 1;
}

message Embedding {
  string user_id = 1;
  repeated float vector = 2;
}

message EmbedResponse {
  repeated Embedding embeddings = 1;
}
